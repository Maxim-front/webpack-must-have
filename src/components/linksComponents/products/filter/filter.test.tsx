import React from "react";
import renderer from "react-test-renderer";
import Filter from "./filter";

it("renders correctly", () => {
  const onChange = (value: string, name?: string) => {
    console.log(value, name);
  };

  const component = renderer.create(<Filter onchange={onChange} platform="platform" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
