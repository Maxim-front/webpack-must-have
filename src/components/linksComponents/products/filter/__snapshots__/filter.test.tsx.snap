// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    platform
  </h3>
  <hr />
  <h4>
    Sort
  </h4>
  <hr />
  <div>
    <div>
      <p>
        Criteria
      </p>
      <select
        defaultValue="rating"
        name="sortByCriteria"
        onChange={[Function]}
      >
        <option
          value="rating"
        >
          Rating
        </option>
        <option
          selected={true}
          value="price"
        >
          Price
        </option>
      </select>
    </div>
    <div>
      <p>
        Type
      </p>
      <select
        defaultValue="asc"
        name="sortByType"
        onChange={[Function]}
      >
        <option
          selected={true}
          value="asc"
        >
          ascending
        </option>
        <option
          value="desc"
        >
          descending
        </option>
      </select>
    </div>
  </div>
  <div>
    <h4>
      Genres
    </h4>
    <hr />
    <div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="genres"
          onChange={[Function]}
          type="radio"
          value="All genres"
        />
        <span>
          All genres
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="genres"
          onChange={[Function]}
          type="radio"
          value="Shooter"
        />
        <span>
          Shooter
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="genres"
          onChange={[Function]}
          type="radio"
          value="Arcade"
        />
        <span>
          Arcade
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="genres"
          onChange={[Function]}
          type="radio"
          value="Survive"
        />
        <span>
          Survive
        </span>
      </div>
    </div>
  </div>
  <div>
    <h4>
      Age
    </h4>
    <hr />
    <div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="age"
          onChange={[Function]}
          type="radio"
          value="All ages"
        />
        <span>
          All ages
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="age"
          onChange={[Function]}
          type="radio"
          value="3"
        />
        <span>
          3
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="age"
          onChange={[Function]}
          type="radio"
          value="6"
        />
        <span>
          6
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="age"
          onChange={[Function]}
          type="radio"
          value="12"
        />
        <span>
          12
        </span>
      </div>
      <div>
        <label
          htmlFor="finput"
        />
        <input
          name="age"
          onChange={[Function]}
          type="radio"
          value="18"
        />
        <span>
          18
        </span>
      </div>
    </div>
  </div>
</div>
`;
